src/dotnet/Program.cs

```diff
@@ -24,7 +24,7 @@ namespace Microsoft.DotNet.Cli
 {
     public class Program
     {
-        private static readonly string ToolPathSentinelFileName = $"{Product.Version}.toolpath.sentinel";
+        private static readonly string ToolPathSentinelFileName = $"{SdkProduct.Version}.toolpath.sentinel";
```

Comment:
I don't get why we have both Product and SdkProduct, if we're using SdkProduct exclusively at this stage. Can't we just change the implementation of Product.Version to use the .version file. 

If/when we surface the CLI version, the names 'Product' and 'SdkProduct' aren't very suggestive of the actual difference.  I'd wait until we actually do that to have both and then we can pick appropriate names.